src/fast_check/transform_dts.rs

```diff
@@ -0,0 +1,730 @@
+            ))
+          }
+          // Keep all these
+          ModuleDecl::TsImportEquals(_)
+          | ModuleDecl::TsNamespaceExport(_)
+          | ModuleDecl::TsExportAssignment(_) => {
+            new_items.push(ModuleItem::ModuleDecl(module_decl.clone()));
+          }
+          _ => {}
+        },
+        ModuleItem::Stmt(stmt) => {
+          if let Stmt::Decl(decl) = stmt {
+            match decl {
+              Decl::TsInterface(_)
+              | Decl::TsTypeAlias(_)
+              | Decl::TsEnum(_)
+              | Decl::TsModule(_) => {
+                new_items.push(ModuleItem::Stmt(stmt.clone()));
+              }
+              _ => {}
+            }
+          }
+        }
+      }
+    }
+
+    module.body = new_items;
+    Ok(module)
+  }
+
+  fn expr_to_ts_type(&mut self, expr: Expr, as_const: bool) -> Option<TsType> {
+    match expr {
+      Expr::Array(arr) => {
+        let mut elem_types: Vec<TsTupleElement> = vec![];
+
+        for elems in arr.elems {
+          if let Some(ts_expr) =
+            elems.and_then(|item| self.expr_to_ts_type(*item.expr, as_const))
+          {
+            elem_types.push(ts_tuple_element(ts_expr));
+          }
+        }
+
+        Some(ts_readonly(TsType::TsTupleType(TsTupleType {
+          span: DUMMY_SP,
+          elem_types,
+        })))
+      }
+      Expr::Object(obj) => {
+        let mut members: Vec<TsTypeElement> = vec![];
+
+        for item in obj.props {
+          match item {
+            PropOrSpread::Prop(prop_box) => {
+              let prop = *prop_box;
+              match prop {
+                Prop::KeyValue(key_value) => {
+                  let (key, computed) = match key_value.key {
+                    PropName::Ident(ident) => (Expr::Ident(ident), false),
+                    PropName::Str(str_prop) => {
+                      (Expr::Lit(Lit::Str(str_prop)), false)
+                    }
+                    PropName::Num(num) => (Expr::Lit(Lit::Num(num)), true),
+                    PropName::Computed(computed) => (*computed.expr, true),
+                    PropName::BigInt(big_int) => {
+                      (Expr::Lit(Lit::BigInt(big_int)), true)
+                    }
+                  };
+
+                  let init_type = self
+                    .expr_to_ts_type(*key_value.value, as_const)
+                    .map(type_ann);
+
+                  members.push(TsTypeElement::TsPropertySignature(
+                    TsPropertySignature {
+                      span: DUMMY_SP,
+                      readonly: true,
+                      key: Box::new(key),
+                      computed,
+                      optional: false,
+                      init: None,
+                      params: vec![],
+                      type_ann: init_type,
+                      type_params: None,
+                    },
+                  ));
+                }
+                // TODO: Requires type resolving, skip for now
```

Comment:
Should we surface a diagnostic here to say we can't properly create a dts file? Also, does this require type resolving? Can we do `ident` -> `ident: typeof ident` or something similar?